06week/dodgeBall.js

```diff
@@ -0,0 +1,373 @@
+  //create a list of it's children element
+  let pULchildren = preUL.children
+  // console.log('pul',pULchildren)
+  let matchingID 
+
+  //look at the items moved to your new list
+  newArray.forEach((element) => {
+    matchingID = element.id
+    let findMatch = pULchildren.namedItem(matchingID)
+    // console.log('find a match?', findMatch)
+        //if the id for someone in your new array of players, then delete that li
+    if(findMatch) {
+      // console.log('and the match was ',findMatch, 'previousArray~~', previousArray, `& element id this go is:`, matchingID )
+      // lastDIV.appendChild(preUL)
+      // thatIndex = previousArray.indexOf(`${element.id}`)
+      // previousArray.splice(thatIndex, 1)
+      return preUL.removeChild(findMatch)
+    } 
+    else if (!findMatch) {
+      // console.log('there should not be a match after it was removed')
+      return previousDIV
+    }
+    return appending(previousDIV, previousUL)
+  })
+  // console.log( 'HTML?',pULchildren.namedItem(2))
+  // console.log('completed the removal')
+    //we need to remove the player from the previous array so that they don't keep appearing on the browser   
+    if(!previousArray) {
+      // console.log('skip this step, you did not provide a previous array, so we guess you want to keep it where it is')
+    } else if (
+      previousArray.forEach((element, index) => {
+        // console.log('side my else if to remove the repeat element', element)
+        if(element.id === matchingID) {
+          // console.log('you find the item in the array!')
+          return previousArray.splice(index, 1)
+        }
+        return previousArray
+      }) 
+    ) {
+      console.log('found in previous array! Now you have to remove it', previousArray)
+      return previousArray
+    }
+}
+
+//take the array of people and put it in this list w/ the player components added. 
+const listOfPlayers = []
+
+const makePlayer = (id) => {
+  // console.log(`li ${id} was clicked!`)
+  //this creates a copy of the individual with a matching id to the list of players array. 
+  arrOfPeople.map(person => {
+    //pull in the whole object, but only add those individual to the listOfPlays, if their id was provided 
+    // console.log('look at the whole object,', person)
+    if(person.id === id) {
+      // console.log('first if, inside makePlayer?')
+      //at this point, we want to create a new object which includes the player constructor and the person object from the arrayOfPeople
+          //we are assuming all of these when they sign up. 
+      // console.log(letsPlay)
+      let newPlayer = [new Player(false, false, false, false,0)]
+      let assignPlayer = Object.assign(person,{newPlayer})
+      // console.log(listOfPlayers, 'this the list of players')
+      // console.log('merge to objects?',newPlayer)
+    listOfPlayers.push(assignPlayer)
+    }
+  }
+  )
+  // dbUL.innerHTML = ' '
+  listOfPlayers.forEach((person) => {
```

Comment:
the code now, will add every player in the array again to the ul  when a new person is added to the list of players.